### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by danez**
Enables prettier

//cc @vjeux Not many drastic changes here, but in case you want to have a look

//cc @hzoo @yavorsky How should we handle eslint? I disabled one rule `arrow-parens` as it was interfering, should we extend `eslint-config-prettier` in our config?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,5 +92,15 @@
       "src/**/*.js"
     ],
     "babel": "inherit"
+  },
+  "lint-staged": {
+    "src/**/*.js": [
+      "prettier --trailing-comma all --write ",
+      "git add"
+    ],
+    "test/**/*.test.js": [
```
Comment (yavorsky) [inline package.json]: Should we also include `test/helpers/createTestDirectory.js` ?

**Comment by yavorsky** [comment]
eslint-config-prettier is a good idea. I'm going to add it for https://github.com/babel/babel-preset-env/pull/183.

But seems like adding
```json
"arrow-parens": "off",
"indent": "off"
```
is enough to make babel-config prettier-friendly.

File: package.json (hunk 1 above)
Comment (hzoo) [inline package.json]: Do we need to do all of these or can we just run it on all .js files?

File: src/fs-cache.js [hunk 2]
```diff
@@ -57,13 +56,12 @@ const write = function(filename, result, callback) {
   const content = JSON.stringify(result);
 
   return zlib.gzip(content, function(err, data) {
-    if (err) { return callback(err); }
+    if (err) return callback(err);
```
Comment (hzoo) [inline src/fs-cache.js]: I like the braces but ok

File: package.json (hunk 1 above)
Comment (yavorsky) [inline package.json]: @hzoo I think the only reason why can't do this is that running on all .js files will affect fixtures

File: src/fs-cache.js (hunk 2 above)
Comment (vjeux) [inline src/fs-cache.js]: Note that this is not prettier doing. Prettier will keep the {} as they were inputted. But, if there are braces, the expression will be on its own line, if there are no braces, the expression is inline like it is here.

**Comment by existentialism** [comment]
It might be fine to just remove the few style rules we have in eslint-config-babel?

File: src/fs-cache.js (hunk 2 above)
Comment (hzoo) [inline src/fs-cache.js]: Ok I didn't think so

**Pull request merged**